apps/prairielearn/src/lib/manualGrading.js

```diff
@@ -34,4 +51,531 @@ module.exports.nextUngradedInstanceQuestionUrl = async (
+        rubric_id,
+        computed_points,
+        adjust_points: adjust_points || 0,
+        rubric_items: JSON.stringify(rubric_items || []),
+      })
+    ).rows[0];
+
+    return {
+      id: rubric_grading_result.id,
+      computed_points,
+      replace_auto_points: rubric_data.replace_auto_points,
+    };
+  });
+}
+
+/** Manually updates the score of an instance question.
+ * @param {number} assessment_id - The ID of the assessment associated to the instance question. Assumed to be safe.
+ * @param {number} instance_question_id - The ID of the instance question to be updated. May or may not be safe.
+ * @param {number|null} submission_id - The ID of the submission. Optional, if not provided the last submission if the instance question is used.
+ * @param {string|null} check_modified_at - The value of modified_at when the question was retrieved, optional. If provided, and the modified_at value does not match this value, a grading job is created but the score is not updated.
+ * @param {Object} score - The score values to be used for update.
+ * @param {number|string} [score.manual_points] - The manual points to assign to the instance question.
+ * @param {number|string} [score.manual_score_perc] - The percentage of manual points to assign to the instance question.
+ * @param {number|string} [score.auto_points] - The auto points to assign to the instance question.
+ * @param {number|string} [score.auto_score_perc] - The percentage of auto points to assign to the instance question.
+ * @param {number|string} [score.points] - The total points to assign to the instance question. If provided, the manual points are assigned this value minus the question's auto points.
+ * @param {number|string} [score.score_perc] - The percentage of total points to assign to the instance question. If provided, the manual points are assigned the equivalent of points for this value minus the question's auto points.
+ * @param {Object} [score.feedback] - Feedback data to be provided to the student. Freeform, though usually contains a `manual` field for markdown-based comments.
+ * @param {Object} [score.partial_scores] - Partial scores associated to individual elements. Must match the format accepted by individual elements. If provided, auto_points are computed based on this value.
+ * @param {Object} [score.manual_rubric_data] - Rubric items associated to the grading of manual points. If provided, overrides manual points.
+ * @param {Object} [score.manual_rubric_data.rubric_id] - Rubric ID to use for manual grading.
+ * @param {AppliedRubricItem[]} [score.manual_rubric_data.applied_rubric_items] - Applied rubric items.
+ * @param {number | string} [score.manual_rubric_data.adjust_points=0] - number of points to add (positive) or subtract (negative) from the total computed from the items.
+ * @param {number} authn_user_id - The user_id of the logged in user.
+ * @returns {Promise<Object>}
+ */
+async function updateInstanceQuestionScore(
```

Comment:
This function (and previously the sproc) was mainly used for manual grading (UI, upload, or manually setting the points in the assessment instance page). I think the placement here makes sense. IMHO the fact that it is able to change the auto-points is not relevant, since it would be used in manual updates of those points.

Reply:
Ok, sounds good, thanks!